muesli/models.py

```diff
@@ -579,10 +637,116 @@ class Grading(Base):
     examiner_id = Column(Text)
     exams = relationship(Exam, secondary=grading_exam_table, backref = "gradings", order_by=Exam.id)
 
+
 class StudentGrade(Base):
     __tablename__ = 'student_grades'
     grading_id = Column('grading', Integer, ForeignKey(Grading.id), nullable=False, primary_key=True)
     grading = relationship(Grading, backref=backref('student_grades', lazy='dynamic'))
     student_id = Column('student', Integer, ForeignKey(User.id), nullable=False, primary_key=True)
     student = relationship(User, backref=backref('student_grades', lazy='dynamic'))
     grade = Column(Numeric(precision=2, scale=1), CheckConstraint('grade >= 1.0 AND grade <= 5.0'))
+
+# Schemas
+
+
+class UserSchema(Schema):
+    id = fields.Integer(dump_only=True)
+    email = fields.Email()
+    first_name = fields.String()
+    last_name = fields.String()
+    matrikel = fields.String()
+    birth_date = fields.String()
+    birth_place = fields.String()
+    subject = fields.String()
+    #TODO Rest
+
+#TODO MANY???!?
+    @post_load()
+    def get_user(self, data):
+        # TODO check also by mail because it's unique
+        usr = self.context['session'].query(User).filter(User.email == data["email"]).one()
+        if usr is None:
+            raise ValidationError("User not found")
+        return usr
+
+
+class ExamSchema(Schema):
+    id = fields.Integer(dump_only=True)
+    lecture_id = fields.Integer(dump_only=True)
+    name = fields.String()
+    category = fields.String()  # TODO verify
+    url = fields.Url()
+
+
+class TutorialSchema(Schema):
+    id = fields.Integer(dump_only=True)
+    place = fields.String()
+    time = fields.Method("get_time")
+    max_students = fields.Integer()
+    tutor = fields.Nested( UserSchema, only=allowed_attributes.user())
+    comment = fields.String()
+    students = fields.Nested(UserSchema, many=True, only=allowed_attributes.user())
+
+    # CORRECT ?! TODO
+    exams = fields.Nested(ExamSchema, many=True, only=["id", "name"])
+    
+    student_count = fields.Method("get_student_num")
+
+    def get_time(self, obj):
+        return obj.time.__html__()
+
+    def get_student_num(self, obj):
+        return obj.students.count()
+
+
+class LectureSchema(Schema):
+    id = fields.Integer(dump_only=True)
+    assistants = fields.Nested(UserSchema, required=True, many=True, only=allowed_attributes.user())
+    name = fields.String(required=True)
+    type = fields.String()  # per default lecture
+    term = fields.Method("get_term", deserialize="load_term")
+    lsf_id = fields.String()
```

Comment:
I don't like, that you implement the same sort of validation mechanism that is already in forms.py. Yours seems to be better, but please combine these two somehow. For example in this line, you accept an lsf_id of 25 characters, but in forms.py the limit is 20.